components/nimbus/src/lib.rs

```diff
@@ -278,18 +299,44 @@ impl NimbusClient {
         Ok(())
     }
 
+    /**
+     * Calculates the active_experiments based on current enrollments for the targeting attributes.
+     */
+    fn update_ta_active_experiments(
+        &self,
+        db: &Database,
+        writer: &Writer,
+        state: &mut MutexGuard<InternalMutableState>,
+    ) -> Result<()> {
+        let enrollments_store = db.get_store(StoreId::Enrollments);
+        let prev_enrollments: Vec<enrollment::ExperimentEnrollment> =
+            enrollments_store.collect_all(writer)?;
+
+        let mut set = HashSet::<String>::new();
+        for ee in prev_enrollments {
+            if let EnrollmentStatus::Enrolled { .. } = ee.status {
```

Comment:
Could you briefly explain what this is doing? I'm familiar with `if let` but I'm not sure what `EnrollmentStatus::Enrolled { .. }` is doing. Is it creating an instance of the Enrolled status and comparing it to the status of the enrollments from the store?

Reply:
Ah yes.

The `if let` takes a pattern and a scrutinee operand, as syntactic sugar for a more verbose `match` expression.

```rs
match ee.status {
   EnrollmentStatus::Enrolled { .. } => { /* blah */ }
   _ => { /* else nothing */ }
}
```

The _pattern_ in this case is [destructuring an enum with a Rest Pattern](https://doc.rust-lang.org/reference/patterns.html#destructuring), so we're saying: "if status is enrolled, regardless of the reason".

Reply:
Ah very cool, thanks for the explanation!